lib/tasks/rubocop.rake

```diff
@@ -0,0 +1,45 @@
+#--
+# Yast rake
+#
+# Copyright (C) 2018 SUSE LLC
+#   This library is free software; you can redistribute it and/or modify
+# it only under the terms of version 2.1 of the GNU Lesser General Public
+# License as published by the Free Software Foundation.
+#
+#   This library is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+# details.
+#
+#++
+
+# run Rubocop in parallel
+# @param params [String] optional Rubocop parameters
+def run_rubocop(params = "")
+  # how it works:
+  # 1) get the list of inspected files by Rubocop
+  # 2) shuffle it randomly (better would be evenly distribute them according to
+  #    the analysis complexity but that is hard to evaluate and even simply
+  #    distributing by file size turned out to be ineffective and slower than
+  #    a simple random shuffling)
```

Comment:
what is reason for shuffling? is it in average more efficient? any measure that it make sense?

Reply:
I tested that locally in yast2-storage-ng (627 files to check) and the average times are:
- No shuffling: 15.8s
- With shuffling: 11.7s

That's ~+35% speed up, quite a lot for such a simple and stupid change. Of course, it depends on the number of tests and their naming. I think it should not harm in general, but in some cases it might help.

The best solution would be similar to the `parallel_rspec` solution - it records the time for each file, but that would be not trivial to implement and the gain would not be big. On my machine the time difference between the first and the last finished job is like 2 seconds, so we would go from ~12s -> ~11s, that's about 10% improvement. Not much for the work required, this simple solution helped with 35%...

Reply:
please try it also in different project like bootloader, yast2 or any slighly bigger yast repo that has rubocop enabled. What works nicely for storage-ng can be in average worse or same in others.

Reply:
OK, will do.

Reply:
Here are the result for all YaST packages, the table contains the package name, the time with the default order and the time with random ordering (in seconds). Each test was executed 3x, the table contains the average time.

|                           |   default |    random | difference |
| ------------------------- | ---------:| ---------:| ----------:|
| alternatives              |       1,8 |       1,8 |        0,0 |
| autoyast-integration-test |       2,3 |       2,2 |        0,1 |
| bootloader                |       3,3 |       3,1 |        0,2 |
| caasp                     |       1,9 |       1,9 |        0,0 |
| configuration-management  |       2,1 |       2,2 |       -0,1 |
| docker                    |       1,5 |       1,6 |       -0,1 |
| firewall                  |       2,0 |       2,0 |        0,0 |
| ftp-server                |       2,2 |       2,3 |       -0,1 |
| helper_scripts            |       0,1 |       0,1 |        0,0 |
| installation              |       5,5 |       5,3 |        0,2 |
| journal                   |       1,9 |       1,9 |        0,0 |
| migration                 |       1,8 |       1,8 |        0,0 |
| network                   |       7,3 |       6,1 |        1,2 |
| nfs-client                |       2,1 |       2,1 |        0,0 |
| ntp-client                |       2,2 |       2,4 |       -0,2 |
| packager                  |       7,4 |       7,1 |        0,3 |
| partitioner               |       0,1 |       0,1 |        0,0 |
| rake-ci                   |       1,3 |       1,3 |        0,0 |
| rake                      |       1,6 |       1,6 |        0,0 |
| registration              |       4,1 |       4,0 |        0,1 |
| rubocop-yast              |       0,2 |       0,2 |        0,0 |
| ruby-bindings             |       2,8 |       3,0 |       -0,2 |
| storage-ng                |      17,8 |      14,0 |        3,8 |
| s390                      |       3,1 |       3,3 |       -0,2 |
| yast2                     |      14,8 |      12,7 |        2,1 |
| total time                    | 4m35,298s | 4m12,687s |     22,611 |

So it seems it helps for large packages like storage-ng or yast2 where it saves several seconds, for the rest the time is more or less the same. It is slower only in very few cases and if it is slower then the difference is just a fraction of second (0.2s maximum).

The last line contains the complete time for my testing script, that shows that on average for all YaST repositories the random shuffling helps. 

So in my opinion we can enable the random order. If it turns out bad in some cases we can change that later.